### PR TITLE
Added changes for Type Hints (PEP 484)

### PR DESCRIPTION
**Pull request by anupamsingh24**
upload_s3 method was throwing error of incorrect datatype, added List[str] instead of str

### CODE REVIEW

File: bugbug/db.py [hunk 1]
```diff
@@ -37,11 +37,11 @@ def is_registered(path: str) -> bool:
     return path in DATABASES
 
 
-def exists(path):
+def exists(path) -> bool:
```
Comment (marco-c) [inline bugbug/db.py:40]: Type annotations should be added for function arguments too

Comment (anupamsingh24) [reply bugbug/db.py]: ack, have added that in the latest commit

File: bugbug/db.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 logger = logging.getLogger(__name__)
 
 
-def register(path, url, version, support_files=[]):
+def register(path: str, url: str, version: int, support_files=[]):
```
Comment (marco-c) [inline bugbug/db.py:25]: A type annotation is required for `support_files` too, and for the return type

File: bugbug/db.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def is_old_schema(path) -> bool:
     return DATABASES[path]["version"] > prev_version
 
 
-def download_support_file(path, file_name, extract=True) -> bool:
+def download_support_file(path: str, file_name: str, extract=True) -> bool:
```
Comment (marco-c) [inline bugbug/db.py:57]: A type annotation is required for `extract` too

File: bugbug/db.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ def download_support_file(path, file_name, extract=True) -> bool:
 
 
 # Download and extract databases.
-def download(path, support_files_too=False, extract=True) -> bool:
+def download(path: str, support_files_too=False, extract=True) -> bool:
```
Comment (marco-c) [inline bugbug/db.py:82]: Type annotations are required for `support_files_too` and `extract` too

File: bugbug/db.py [hunk 5]
```diff
@@ -106,15 +106,15 @@ def download(path, support_files_too=False, extract=True) -> bool:
...
+def upload(path: str) -> None:
     support_files_paths = [
         os.path.join(os.path.dirname(path), support_file_path)
         for support_file_path in DATABASES[path]["support_files"]
     ]
     utils.upload_s3([f"{path}.zst", f"{path}.version"] + support_files_paths)
 
 
-def last_modified(path):
+def last_modified(path: str):
```
Comment (marco-c) [inline bugbug/db.py:117]: The return type is missing

**Comment by marco-c** [comment]
@anupamsingh24 are you still interested on working on this?

**Comment by marco-c** [comment]
Feel free to reopen if you're still interested.

**Pull request closed without merging**